apps/teamsjs-diagnostic-app/src/components/sample/Scenario1.tsx

```diff
@@ -0,0 +1,155 @@
+    if (storedLogs) {
+      setLogStatements(JSON.parse(storedLogs));
+    }
+  }, []);
+
+  const generateVerticalBoxes = () => {
+    const filteredApis = apiComponents.filter(api =>
+      api.title.toLowerCase().includes(searchQuery.toLowerCase())
+    );
+
+    const options = ['Option 1', 'Option 2', 'Option 3'];
+    return filteredApis.map((api, index) => (
+      <div key={index} className="vertical-box">
+        <div className="api-container">
+          <div className="api-header">{api.title}</div>
+          <div className="dropdown-menu">
+            <label htmlFor={`select-${index}`} className="sr-only">
+              Select an option for API {index}
+            </label>
+            <select id={`select-${index}`} className="box-dropdown">
+              {options.map((option, optionIndex) => (
+                <option key={optionIndex} value={option}>
+                  {option}
+                </option>
+              ))}
+            </select>
+          </div>
+        </div>
+      </div>
+    ));
+  };
+
+  return (
+    <div>
+      <h2>App Initialization Scenario</h2>
+      <p>Click the button to run the app initialization scenario.</p>
+      <div className="scenario-container">
+        <div className="scenario1-box">
+          <button
+            className="scenario1-button"
+            onClick={runAppInitializationScenario}
+            type="button"
+            data-testid="run-scenario-button"
+          >
+            Run Scenario
+          </button>
+          <div className="api-section">
+            <div className="api-header">APIs Being Run:</div>
+            <div className="vertical-box-container">
+              <div className="vertical-box">
+                <span className="box-title">1. app</span>
+              </div>
+              <div className="vertical-box">
+                <span className="box-title">2. authentication</span>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="scenario2-container">
+          <div className="scenario2-header">
+            <h2>Custom Scenario</h2>
+            <p>Click the button to run your custom scenario.</p>
+          </div>
+          <div className="custom-scenario-box">
+            <button className="scenario1-button">Run Scenario</button>
+            <button className="set-scenario-button">+</button>
+            <div className="api-section">
+              <div className="api-header">APIs Being Run:</div>
+            </div>
+          </div>
+        </div>
+
+        <div className="all-api-container">
+          <input
+            type="text"
+            className="search-bar"
+            placeholder="Search APIs..."
+            value={searchQuery}
+            onChange={(e) => setSearchQuery(e.target.value)}
+          />
+          <div className="all-api-box">{generateVerticalBoxes()}</div>
+        </div>
+      </div>
+
+      <h2>Log Console</h2>
```

Comment:
Consider making things like this separate react components in their own file. You already have a LoggerUtility.tsx file with no UI, if you made that a react component it could contain all of the logging related logic and UI.

Reply:
You could actually do this for other parts of this file as well, "Custom Scenario" for example could make sense as it's own react component. Right now, this file seems like a collection of multiple different things, but if you got this down to just things that are used in "Scenario 1" that would be great!